src/Interpreters/MutationsInterpreter.cpp

```diff
@@ -178,12 +181,12 @@ bool isStorageTouchedByMutations(
         if (command.type == MutationCommand::APPLY_DELETED_MASK)
         {
             if (source_part->hasLightweightDelete())
-                return true;
+                return all_rows;
```

Comment:
AFAIU in this case we don't know how many rows are touched.
It seems to be safer to return not `all_rows` but `some_rows` (i.e `{.any_rows_affected = true, .all_rows_affected = false}`). Otherwise it might be prone to bug if we decide to check that `APPLY_DELETED_MASK` is the only command.